app/models/concerns/pre_sale_prices/line_item_pre_saleable.rb

```diff
@@ -19,7 +19,7 @@ def copy_pre_sale_price
     def update_pre_sale_price
       variant_pre_sale_price = variant.pre_sale_price_in(variant.currency).amount
 
-      if variant_pre_sale_price > 0
+      if variant_pre_sale_price.present? and variant_pre_sale_price > 0
```

Comment:
Weird to have the #amount method on this return a nil, is that column in the db nullable? Should it have a default value of 0?